enhancements/installer/component-selection.md

```diff
@@ -94,233 +95,183 @@ N/A - no api extensions are being introduced in this proposal
+  This list will remain the same no matter which version of OpenShift is installed.
+* `vCurrent`: the recommended set of optional capabilities to enable for the cluster current version of OpenShift.
+  For a 4.11 cluster, this matches `v4.11`.  We may add or remove capabilities from this set in 4.12 clusters.
 
-From the remaining set of resources:
-3) if any resource is part of an explicitly excluded capability, remove it
-4) if any resource is part of an explicitly included capabilitity, keep it
-5) if the inclusionDefault is Exclude, remove all remaining resources(i.e. not covered by (4)) that are part of a capability
+Admins who want to pre-emptively exclude future capabilities can use `None` or `v4.11` or another `baselineCapabilitySet` which will not evolve with time.
 
+We may decide to add additional sets like `v4.12` or `myGreatUseCase` in the future, but do not commit to doing so.
 
-The `inclusionDefault` field (carried through to the CVO) allows an Admin to choose what behavior they want for new
-capabilities that might come as part of a future upgrade (either have the new capability installed during
-the upgrade, or exclude it).  The `inclusionDefault` field itself will default to `Include`.
+#### Manifest annotations
 
-Specifying the same capability as an `include` and an `exclude` is an error condition.
+Formalizing a concept of a "capability" annotation which allows a given resource to be excluded based
+on installer input. For example the console related resources could be annotated as
 
-Specifying a capability name that isn't represented in the payload is an error condition.  Admins who want to
-pre-emptively exclude future capabilities can use `inclusionDefault: Exclude` and then explicitly include
-anything they want the cluster to have.
+```yaml
+annotations:
+  capability.openshift.io/name: console
+```
 
-Examples of candidate components to be treated as capabilities:
+If a resource participates in multiple capabilities, it should specify all capabilities as plus-sign separated
+values in the annotation(e.g. `capability.openshift.io/name=console+monitoring`).  The resource will only be 
+included if all the capabilities on the resource are enabled.  This allows a component
+to define a `ServiceMonitor` which is part of both the `monitoring` capability and the `Foo` capability.  If
+either of those capabilities are excluded, the `ServiceMonitor` should not be created.  (If `monitoring` is excluded
+then no ServiceMonitors should be created, if only `Foo` is excluded, then only Foo's ServiceMonitors should be
+excluded while ServiceMonitors associated with other capabilities would not be).
 
-* console
-* imageregistry
-* samples
-* cluster baremetal operator
-* olm/marketplace
-* csi-*
-* insights
-* monitoring
-* ???
+To determine which manifests are managed for a given cluster, the CVO will:
 
+1. Start with all the manifests in the payload.
+2. Remove all the manifests that have no annotations.
```

Comment:
[We do][1].  Although if you wanted, you could read this step as a special case of the cluster-profile exclusion.

[1]: https://github.com/openshift/library-go/blob/f2c20ff2ad2771dd39c1482658f18c2587da6469/pkg/manifest/manifest.go#L123-L126

Reply:
oh right, because they *all* have at least the cluster profile annotation.  Do we even have any manifests that don't have an annotation?  I assume not since it would do nothing.

Reply:
(might be worth clarifying that most/all manifests do have at least one annotation, due to cluster profiles.  Otherwise this reads like we're going to remove all the manifests that don't have a capability annotation)

Reply:
> Do we even have any manifests that don't have an annotation?

We do: [CredentialRequests for platforms where there is no in-cluster cloud-cred-operator management][1].

> ... might be worth clarifying that most/all manifests do have at least one annotation, due to cluster profiles...

Seemed simpler to just drop this line, rely on the profiles check listed here to cover unannotated manifests, and treat the current `Manifest.Include` check as an internal optimization, so I've pushed ed81a86b -> aea25fe1 to drop this line.

[1]: https://issues.redhat.com/browse/CCO-210